src/Datadog.Trace/Configuration/GlobalSettings.cs

```diff
@@ -184,5 +223,21 @@ private static bool TryLoadHostingEnvironmentPath(out string hostingPath)
             hostingPath = default;
             return false;
         }
+
+        private static void LoadVendorFactoryConfigurator()
+        {
+            if (Source.VendorPluginLoaded)
+            {
+                var factoryInterface = typeof(IFactoryConfigurator);
+                var factoryType = Source.VendorPlugin
+                    .GetTypes()
+                    .FirstOrDefault(x => factoryInterface.IsAssignableFrom(x));
```

Comment:
I do not like having logic like this in `GlobalSettings` class.

Reply:
Which part of the logic do you not like in this class?

Unrelated, do we only expect one factory per plugin? I could see us getting all types that implement `IFactoryConfigurator` so the factories could configure settings independently.

Reply:
`GlobalSettings` should be responsible for parsing the configuration. It should not invoke logic like loading the plugins. It can load "core" configuration for plugins like where are they stored.

Reply:
> Unrelated, do we only expect one factory per plugin? I could see us getting all types that implement IFactoryConfigurator so the factories could configure settings independently

Good question. 

We can have one "God" interface with all possible extension points. We can always provide an abstract class for ease of use. It would be a "non-SOLID" design however it could be slightly more performance and most importantly - **easier to find out potential extension points for the plugin devs**.

Make an interface per extension point like advised [here](https://blog.ploeh.dk/2014/05/19/di-friendly-framework/). It is a more loose-coupled design.

I do not have any strong opinion here. Internally we would probably still want to have these kind of "little" interfaces. Right now we have one extension point so we can always postpone this discussion and decision until we have more experience. Especially that I believe that as a rule of thumb, we will advise rebuilding the plugin for each version of do OTel Tracer. AFAIK this is what Java AutoInstr plugins do as well. E.g. https://github.com/signalfx/splunk-otel-java/releases/tag/v0.10.0

Reply:
I would prefer "God" interface cause of the performance. There is only single assembly scan and `Activator.CreateInstance` call per plugin dll. Probably should rename that interface also, reflecting more "plugin" in it's name.

Reply:
@pellared makes sense, thanks! We can start with one interface and revisit the decision later if needed